Microsoft.Toolkit.Uwp.UI.Controls/TextBoxMask/TextBoxMask.cs

```diff
@@ -0,0 +1,303 @@
+                    }
+
+                    representationDictionary.Add(key, value);
+                }
+            }
+
+            // TODO: insert generic custom representation
+            textbox.SetValue(RepresentationDictionaryProperty, representationDictionary);
+
+            var displayText = mask;
+            foreach (var key in representationDictionary.Keys)
+            {
+                displayText = displayText.Replace(key, placeHolder);
+            }
+
+            textbox.Text = displayText;
+            textbox.TextChanging -= Textbox_TextChanging;
+            textbox.SelectionChanged -= Textbox_SelectionChanged;
+            textbox.Paste -= Textbox_Paste;
+            textbox.TextChanging += Textbox_TextChanging;
+            textbox.SelectionChanged += Textbox_SelectionChanged;
+            textbox.Paste += Textbox_Paste;
+            textbox.SetValue(OldTextProperty, displayText);
+        }
+
+        private static async void Textbox_Paste(object sender, TextControlPasteEventArgs e)
+        {
+            e.Handled = true;
+            DataPackageView dataPackageView = Clipboard.GetContent();
+            if (!dataPackageView.Contains(StandardDataFormats.Text))
+            {
+                return;
+            }
+
+            var pasteText = await dataPackageView.GetTextAsync();
+            if (string.IsNullOrWhiteSpace(pasteText))
+            {
+                return;
+            }
+
+            var textbox = sender as TextBox;
+            var mask = textbox?.GetValue(MaskProperty) as string;
+            var representationDictionary = textbox?.GetValue(RepresentationDictionaryProperty) as Dictionary<char, string>;
+            if (string.IsNullOrWhiteSpace(mask) ||
+                representationDictionary == null)
+            {
+                return;
+            }
+
+            // to update the textbox text without triggering TextChanging text
+            textbox.TextChanging -= Textbox_TextChanging;
+
+            var oldSelectionStart = (int)textbox.GetValue(OldSelectionStartProperty);
+            var maxLength = pasteText.Length < mask.Length ? pasteText.Length : mask.Length;
+            var textArray = textbox.Text.ToCharArray();
+
+            for (int i = oldSelectionStart; i < oldSelectionStart + maxLength; i++)
+            {
+                var maskChar = mask[i];
+                var selectedChar = pasteText[i - oldSelectionStart];
+
+                // If dynamic character a,9,* or custom
+                if (representationDictionary.ContainsKey(maskChar))
+                {
+                    var pattern = representationDictionary[maskChar];
+                    if (Regex.IsMatch(selectedChar.ToString(), pattern))
+                    {
+                        textArray[i] = selectedChar;
+                    }
+                }
+            }
+
+            textbox.Text = new string(textArray);
+            textbox.SetValue(OldTextProperty, textbox.Text);
```

Comment:
No, because the old text value is used on textChanging not on paste, so the last old text for text change should be after the change.